tests/handlers/compression/test_xz.py

```diff
@@ -0,0 +1,704 @@
+            0,
+            len(XZ_SHA256_CONTENT) * 3,
+            id="sha256_concat",
+        ),
+        pytest.param(
+            XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT,
+            0,
+            len(XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT),
+            id="crc_none_concat_pad",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT,
+            0,
+            len(XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT),
+            id="crc32_concat_pad",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT,
+            0,
+            len(XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT),
+            id="concat_pad",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT,
+            0,
+            len(XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT),
+            id="sha256_concat_pad",
+        ),
+        pytest.param(
+            XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT + NULL_PADDING),
+            id="crc_none_concat_pad2",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT + NULL_PADDING),
+            id="crc32_concat_pad2",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT + NULL_PADDING),
+            id="crc64_concat_pad2",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT + NULL_PADDING),
+            id="sha256_concat_pad2",
+        ),
+    ),
+)
+def test_xz_calculate_chunk_prefix_suffix(
+    content: bytes, start_offset: int, expected_length: int
+):
+    handler = XZHandler()
+    fake_file = File.from_bytes(DATA_PADDING + content + DATA_PADDING)
+    start_offset += len(DATA_PADDING)
+    chunk = handler.calculate_chunk(fake_file, start_offset)
+    assert chunk is not None
+    assert chunk.end_offset == start_offset + expected_length
+
+
+def test_xz_calculate_chunk_unique_stream():
+    """We should emit a chunk for XZ streams having the same flags."""
+    formats = [XZ_NONE_CONTENT, XZ_CRC32_CONTENT, XZ_CRC64_CONTENT, XZ_SHA256_CONTENT]
+
+    for first_chunk, second_chunk in itertools.combinations(formats, 2):
+        handler = XZHandler()
+        fake_file = File.from_bytes(first_chunk + second_chunk)
+        chunk = handler.calculate_chunk(fake_file, 0)
+        assert chunk is not None
+        assert chunk.end_offset == len(first_chunk)
+
+
+def test_xz_calculate_chunk_unique_stream_padding():
```

Comment:
`test_xz_calculate_chunk_unique_stream` is the same use parameterize

Reply:
Fixed together.